### PR TITLE
DOC: Improved the docstring of pandas.Series.clip

### PR DESCRIPTION
**Pull request by ReubenMarkham**
```
################################################################################
######################## Docstring (pandas.Series.clip) ########################
################################################################################

Trim values at input threshold(s).

    Truncates values below and above specified thresholds.
    Thresholds can be singular values or array like, and in
the latter case the truncation is performed element-wise
in the specified axis.

Parameters
----------
lower : float or array_like, default None
    Minimum threshold value. All values below this
    threshold will be set to it.
upper : float or array_like, default None
    Maximum threshold value. All values above this 
    threshold will be set to it.
axis : int or string axis name, optional
    Align object with lower and upper along the given axis.
inplace : boolean, default False
    Whether to perform the operation in place on the data
        .. versionadded:: 0.21.0.
args : iterable, optional
    Arguments to pass to the function.
kwargs : mapping, optional
    Keyworded arguments to pass to the function.

See Also
--------
clip_lower : Return copy of the input with values below given value(s) truncated.
clip_upper : Return copy of the input with values above given value(s) truncated.

Returns
-------
clipped : Series

Examples
--------
>>> data = {'0': [9, -3, 0, -1, 5], '1': [-2, -7, 6, 8, -5]}
>>> df = pd.DataFrame(data)
>>> df
        0       1
0       9       -2
1       -3      -7
2       0       6
3       -1      8
4       5       -5

>>> df.clip(-4, 6)
    0   1
0       6       -2
1       -3      -4
2       0       6
3       -1      6
4       5       -4

>>> t = pd.Series([2, -4, -1, 6, 3])
>>> t
0    2
1   -4
2   -1
3    6
4    3
dtype: int64

>>> df.clip(t, t + 4, axis=0)
        0       1
0       6       2
1       -3      -4
2       0       3
3       6       8
4       5       3

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.Series.clip" correct. :)
```

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -5548,53 +5548,74 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
...
-        2  0.027753 -1.000000
-        3  0.230930 -0.679613
-        4  0.500000  0.500000
-
+            0 	1
+        0 	9 	-2
+        1 	-3 	-7
+        2 	0 	6
+        3 	-1 	8
+        4 	5 	-5
```
Comment (stijnvanhoey) [inline pandas/core/generic.py]: alignment of the printed output seems strange. Should align right for each individual column

File: pandas/core/generic.py [hunk 2]
```diff
@@ -5548,53 +5548,74 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
...
+        clip_lower : Clips values below specified threshold(s).
+        clip_upper : Clips values above specified threshold(s).
 
         Returns
         -------
         clipped : Series
 
         Examples
         --------
+        >>> data = {'0': [9, -3, 0, -1, 5], '1': [-2, -7, 6, 8, -5]}
```
Comment (stijnvanhoey) [inline pandas/core/generic.py]: cfr. docstring guide, use actual names for the column names instead of `0`, `1`

File: pandas/core/generic.py [hunk 3]
```diff
@@ -5548,53 +5548,74 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
...
+        3 	-1 	8
+        4 	5 	-5
+
+        >>> df.clip(-4, 6)
+            0 	1
+        0 	6 	-2
+        1 	-3 	-4
+        2 	0 	6
+        3 	-1 	6
+        4 	5 	-4
```
Comment (stijnvanhoey) [inline pandas/core/generic.py]: cfr. comment on alignment

File: pandas/core/generic.py [hunk 4]
```diff
@@ -5548,53 +5548,74 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
         """
         Trim values at input threshold(s).
 
+        Truncates values below and above specified thresholds.
```
Comment (stijnvanhoey) [inline pandas/core/generic.py]: Truncates feels like you throw away the values outside the specified range. I would make it more descriptive by actually saying that values outside the boundaries are reset to the boundary values.

File: pandas/core/generic.py [hunk 5]
```diff
@@ -5548,53 +5548,74 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
...
+                .. versionadded:: 0.21.0.
+        args : iterable, optional
+            Arguments to pass to the function.
+        kwargs : mapping, optional
+            Keyworded arguments to pass to the function.
+
+        See Also
+        --------
+        clip_lower : Clips values below specified threshold(s).
+        clip_upper : Clips values above specified threshold(s).
```
Comment (stijnvanhoey) [inline pandas/core/generic.py]: Clips -> Clip

File: pandas/core/generic.py [hunk 6]
```diff
@@ -5548,53 +5548,74 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
...
+            threshold will be set to it.
         axis : int or string axis name, optional
             Align object with lower and upper along the given axis.
         inplace : boolean, default False
             Whether to perform the operation in place on the data
-                .. versionadded:: 0.21.0
+                .. versionadded:: 0.21.0.
+        args : iterable, optional
+            Arguments to pass to the function.
+        kwargs : mapping, optional
```
Comment (stijnvanhoey) [inline pandas/core/generic.py]: kwargs -> **kwargs

File: pandas/core/generic.py [hunk 7]
```diff
@@ -5548,53 +5548,74 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
...
         upper : float or array_like, default None
+            Maximum threshold value. All values above this
+            threshold will be set to it.
         axis : int or string axis name, optional
             Align object with lower and upper along the given axis.
         inplace : boolean, default False
             Whether to perform the operation in place on the data
-                .. versionadded:: 0.21.0
+                .. versionadded:: 0.21.0.
+        args : iterable, optional
```
Comment (stijnvanhoey) [inline pandas/core/generic.py]: args -> *args

File: pandas/core/generic.py [hunk 8]
```diff
@@ -5548,53 +5548,74 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
...
+            Arguments to pass to the function.
+        kwargs : mapping, optional
+            Keyworded arguments to pass to the function.
```
Comment (stijnvanhoey) [inline pandas/core/generic.py]: See discussion on https://github.com/pandas-dev/pandas/pull/19985/files#diff-248241e14f081931cc805f9d0e137a5fR709

maybe change to:
```
*args, **kwargs 
    Additional keywords have no effect but might be accepted 
    for compatibility with numpy.
```

File: pandas/core/generic.py [hunk 9]
```diff
@@ -5548,53 +5548,74 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
...
+
+        See Also
+        --------
+        clip_lower : Clips values below specified threshold(s).
+        clip_upper : Clips values above specified threshold(s).
 
         Returns
         -------
         clipped : Series
```
Comment (stijnvanhoey) [inline pandas/core/generic.py]:
```
Returns
----------
`pandas.Series`
    Series with the values outside the clip boundaries replaced
```

**Review by dcanones — APPROVED** [review]
Good!

**Comment by stijnvanhoey** [comment]
@ReubenMarkham why this branch is closed instead of merged?

**Comment by dcanones** [comment]
@stijnvanhoey Hi!, you can find the PR related to this method here: https://github.com/pandas-dev/pandas/pull/20256

Reuben asked me to continue his job and all your suggestions should be already included in the new PR.

**Comment by stijnvanhoey** [comment]
Great! Would have been a pity to not include the work of Reuben. Nice work!

**Pull request closed without merging**